### PR TITLE
Add sparse_dot_topn

### PR DESCRIPTION
**Pull request by oblute**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by oblute** [comment]
@rluria14 @benhuff agree to be maintainers?

File: recipes/sparse-dot-topn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - pip
+    - python
+    - numpy
+    - cython
+    - setuptools >=18.0
+  run:
+    - python
+    - numpy
+    - cython
+    - setuptools >=18.0
```
Comment (scopatz) [inline recipes/sparse-dot-topn/meta.yaml]: Are setuptools and cython really runtime deps?

File: recipes/sparse-dot-topn/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - {{ compiler('cxx') }}
+  host:
+    - pip
+    - python
+    - numpy
+    - cython
+    - setuptools >=18.0
+  run:
+    - python
+    - numpy
```
Comment (scopatz) [inline recipes/sparse-dot-topn/meta.yaml]: Also, please follow this pattern if building against numpy: https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=numpy#building-against-numpy

**Comment by scopatz** [comment]
Thanks for putting this in @oblute!

File: recipes/sparse-dot-topn/meta.yaml (hunk 1 above)
Comment (oblute) [reply recipes/sparse-dot-topn/meta.yaml]: @scopatz they were in the requirements.txt, but it builds successfully without them. thanks!

File: recipes/sparse-dot-topn/meta.yaml (hunk 2 above)
Comment (oblute) [reply recipes/sparse-dot-topn/meta.yaml]: @scopatz got it - fixed below. thanks!

**Comment by oblute** [comment]
@scopatz @conda-forge/help-python this recipe is ready for final review

**Pull request merged**